fsmonitor.c

```diff
@@ -71,6 +88,8 @@ void write_fsmonitor_extension(struct strbuf *sb, struct index_state *istate)
 	uint32_t ewah_size = 0;
 	int fixup = 0;
 
+	assert(istate->fsmonitor_dirty->bit_size <= istate->cache_nr);
```

Comment:
Again, a `BUG()` might make it easier to catch these things early, should they hit production.